app/src/main/java/pm/gnosis/heimdall/ui/tokens/payment/PaymentTokensActivity.kt

```diff
@@ -66,6 +73,20 @@ class PaymentTokensActivity : ViewModelActivity<PaymentTokensContract>() {
                     R.string.fee
             }
         )
+
+        if (nextAction != null) {
+            payment_token_bottom_panel.visible(nextAction != null)
+            viewModel.paymentToken.observe(this, Observer {
+                payment_token_bottom_panel.forwardLabel = getString(R.string.pay_with, it.symbol)
+            })
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        disposables += payment_token_bottom_panel.forwardClicks.subscribeBy {
+            startActivity(intent.getParcelableExtra(EXTRA_NEXT_ACTION) as Intent)
```

Comment:
I would feel safer if we check if `EXTRA_NEXT_ACTION` exists or returns `null` else it will crash here

Reply:
if it is null, bottom panel is gone. But extra check won't hurt